sdk/src/wallet/utxo.ts

```diff
@@ -69,20 +71,21 @@ export async function createBitcoinPsbt(
     const addressInfo = getAddressInfo(fromAddress);
 
     // TODO: possibly, allow other strategies to be passed to this function
-    const utxoSelectionStrategy = 'default';
+    const utxoSelectionStrategy: SelectionStrategy = 'default';
 
     if (addressInfo.network === 'regtest') {
         throw new Error('Bitcoin regtest not supported');
     }
 
     // We need the public key to generate the redeem and witness script to spend the scripts
-    if (addressInfo.type === (AddressType.p2sh || AddressType.p2wsh)) {
+    if (addressInfo.type === AddressType.p2sh || addressInfo.type === AddressType.p2tr) {
```

Comment:
The answer is [here](https://github.com/bob-collective/bob/pull/408#discussion_r1832385126)

Reply:
`if (addressInfo.type === (AddressType.p2sh || AddressType.p2wsh)) {` more detailed reply

this `if` statement only checks if `addressInfo.type === AddressType.p2sh`. The reason is that `(AddressType.p2sh || AddressType.p2wsh)` this expression always resolves as `AddressType.p2sh` -- result of `or` operator is the first non-falsy value. I also notices that in tests for utxo file `publicKey` was not provided in case if address type is `p2wsh`.

Reply:
> Ok so `AddressType.p2wsh` was never actually checked for. The question is, though, _should_ it be checked for?

Will add this check. It should not break anything.

Reply:
But based on this comment it looks like p2wsh have yet to be supported https://github.com/bob-collective/bob/blob/154b7339a31cb771b91f576fc8cf9587e2bf25fd/sdk/src/wallet/utxo.ts#L194-L195